pkg/policy/rules.go

```diff
@@ -252,3 +253,29 @@ egressLoop:
 
 	return egressDecision
 }
+
+// analyzeWhetherRulesSelectEndpoints iterates over a given list of rules to
+// update the cache within the rule which determines whether or not the given
+// identity is selected by that rule. If a rule in the list does select said
+// identity, it is added to epIDSet. Signals to the given WaitGroup that
+// all rules have been parsed in relation to said identity.
+func (rules ruleSlice) analyzeWhetherRulesSelectEndpoint(identityConsumer IdentityConsumer, epIDSet *IDSet, wg *sync.WaitGroup) {
```

Comment:
The name of this function suggests that its primary purpose is to determine whether the endpoint is selected by the received rules structure. However if this were the case, I think the loop below could be cut short.

Rather, I think that the real intent of this function is to `updateEndpointsCache`, with a side effect of determining whether the endpoint is selected or not (really just a boolean result, but this result is instead being pushed into `IDSet` which is expected to be shared across many goroutines).

Is this a fair observation? I think this is just a naming thing, but it would clarify the cost of this function for callers. I also called out the locking part of this with the result below.

Reply:
I renamed the function, let me know what you think.